javacutil/src/org/checkerframework/javacutil/AnnotationBuilder.java

```diff
@@ -153,6 +156,28 @@ public AnnotationMirror build() {
         return new CheckerFrameworkAnnotationMirror(annotationType, elementValues);
     }
 
+    /**
+     * Copies every element value from the given annotation. If an element in the given annotation
+     * doesn't exist in the annotation to be built, an error is raised unless the element is
+     * specified in {@code ignorableElements}.
+     *
+     * @param valueHolder the annotation that holds the values to be copied
+     * @param ignorableElements the elements that can be safely dropped
+     */
+    public void copyElementValuesFromAnnotation(
+            AnnotationMirror valueHolder, String... ignorableElements) {
+        Set<String> ignorableElementsSet = new HashSet<>(Arrays.asList(ignorableElements));
+        for (Map.Entry<? extends ExecutableElement, ? extends AnnotationValue> eltValToCopy :
+                valueHolder.getElementValues().entrySet()) {
+            Name eltNameToCopy = eltValToCopy.getKey().getSimpleName();
+            if (ignorableElementsSet.contains(eltNameToCopy.toString())) {
+                continue;
+            }
+            elementValues.put(findElement(eltNameToCopy), eltValToCopy.getValue());
```

Comment:
I omit the sub type check here as I couldn't find a way to make it pass by passing the key value pair into `setValue` method. Any suggestion here would be great.

Reply:
The code looks much more readable now. I don't think a comment about `setValue` is necessary any more, as you don't duplicate that logic.